DQM/SiStripMonitorClient/src/SiStripTrackerMapCreator.cc

```diff
@@ -363,8 +363,10 @@ void SiStripTrackerMapCreator::paintTkMapFromAlarm(uint32_t det_id, const Tracke
   if(isBad) { rval=255; gval=255; bval = 0; comment << " PCLBadModule ";}
   trackerMap_->setText(det_id, comment.str());
   trackerMap_->fillc(det_id, rval, gval, bval);
-  if(badmodmap && (flag!=0 || isBad)) (*badmodmap)[det_id] = comment.str();
-
+  uint  lay= tTopo->layer(SiStripDetId(det_id));
```

Comment:
@fioriNTU - please put all three lines into the if statement. 'lay' and 'layer' are needed only in the if